### PR TITLE
injector changes for health check, pilot agent take over app readines…

### PR DESCRIPTION
**Pull request by incfly**
CP #9266 in release 1.1.  Pilot agent status server part is already in this branch.

* WIP injector change to modify istio-proxy.

* move out to app_probe.go

* Iterating sidecartmpl to find the statusPort.

* use the same name for ready path.

* Get rewrite work, almost.

* Some clean up on test and check one container criteria.

* fix the injected test file.

* Add inject test for readiness probe itself.

* Add missing added test file.

* fix helm test.

* fix lint.

* update header based finding the port.

* return to previous injected file status.

* fixing TestIntoResource test.

* sed fixing all remaining injecting files.

* handling named port.

* fixing merginge failure.

* remove the debug print.

* lint fixing.

* Apply the suggestions for finding statusPort arg.

* Address comments, regex support more port value format.

* add app_probe_test.go

* add more test.

* merge fix the test.

### CODE REVIEW

**Comment by incfly** [comment]
/cc @liminw 

FYI after this merge, release 1.1 mtls can work with liveness and readiness probe.

File: istioctl/cmd/istioctl/kubeinject.go [hunk 1]
```diff
@@ -375,7 +375,7 @@ func init() {
 	injectCmd.PersistentFlags().StringVar(&imagePullPolicy, "imagePullPolicy", inject.DefaultImagePullPolicy,
 		"Sets the container image pull policy. Valid options are Always,IfNotPresent,Never."+
 			"The default policy is IfNotPresent.")
-	injectCmd.PersistentFlags().IntVar(&statusPort, "statusPort", inject.DefaultStatusPort,
```
Comment (costinm) [inline istioctl/cmd/istioctl/kubeinject.go]: I don't think we can make backward incompatible flag changes. 
Keep old flag, add new if needed.

File: pilot/pkg/kube/inject/app_probe.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	istioProxyContainerName = "istio-proxy"
+)
+
+var (
+	// regex pattern for to extract the pilot agent probing port.
+	// Supported format, --statusPort, -statusPort, --statusPort=15020.
+	statusPortPattern = regexp.MustCompile(fmt.Sprintf(`^-{1,2}%s(=(?P<port>\d+))?$`, StatusPortCmdFlagName))
+)
+
+func rewriteAppHTTPProbe(spec *SidecarInjectionSpec, podSpec *corev1.PodSpec) {
```
Comment (costinm) [inline pilot/pkg/kube/inject/app_probe.go]: For new (and risky) features - please add a flag or env variable to disable the behavior (that's a generic requirement)

File: istioctl/cmd/istioctl/kubeinject.go (hunk 1 above)
Comment (incfly) [reply istioctl/cmd/istioctl/kubeinject.go]: This is the same flag. I just refactor the flag name with `inject.StatusPortCmdFlagName` which equals to "statusPort".

**Comment by incfly** [comment]
I'll do helm install option in a spearate PR on master and then cherrypick that one here.

**Comment by incfly** [comment]
I've already have an PR on top of this one ready to do a helm flag guarded. https://github.com/incfly/istio/compare/release-1.1...incfly:health-flag?expand=1

Let's get this in first and I'll send out the other PR separately. This way we can avoid the conflicts we'll have when merging back to master.

PTAL, @costinm @nmittler

**Pull request merged**